tests/search/includes/classes/test-class-healthjob.php

```diff
@@ -153,4 +155,64 @@ public function test__vip_search_healthjob_process_results_with_wp_error() {
 
 		$stub->process_results( $results );
 	}
+
+	public function test_vip_search_healthjob_is_not_enabled_when_indexing_is_occuring() {
+		add_filter( 'ep_is_indexing', '__return_true' );
+
+		$job = new \Automattic\VIP\Search\HealthJob();
+
+		$enabled = $job->is_enabled();
+
+		$this->assertFalse( $enabled );
+
+		remove_filter( 'ep_is_indexing', '__return_true' );
+	}
+
+	public function test_vip_search_healthjob_is_not_enabled_before_first_index() {
+		add_filter( 'ep_last_sync', '__return_false' );
+
+		$job = new \Automattic\VIP\Search\HealthJob();
+
+		$enabled = $job->is_enabled();
+
+		$this->assertFalse( $enabled );
+
+		remove_filter( 'ep_last_sync', '__return_false' );
+	}
+
+	public function test_vip_search_healthjob_is_enabled_when_expected() {
+		add_filter( 'ep_is_indexing', '__return_false' );
+		add_filter( 'ep_last_sync', '__return_true' );
+
+		// Have to filter the enabled envs to allow `false`, which is the VIP_GO_ENV in tests
+		$enabled_environments = function() {
+			return [ false ];
+		};
+
+		add_filter( 'vip_search_healthchecks_enabled_environments', $enabled_environments );
+
+		$job = new \Automattic\VIP\Search\HealthJob();
+
+		$enabled = $job->is_enabled();
+
+		$this->assertTrue( $enabled );
+
+		remove_filter( 'ep_is_indexing', '__return_false' );
+		remove_filter( 'ep_last_sync', '__return_true' );
+		remove_filter( 'vip_search_healthchecks_enabled_environments', $enabled_environments );
+	}
+
+	/**
+	 * NOTE - needs to come last for the "is_enabled()" tests, b/c it sets the constant
+	 * which causes it to bail early
```

Comment:
This is bound to trip us up in the future. I'd suggest running this job in a separate process to avoid that.

```suggestion
	 * which causes it to bail early
	 *
	 * @runInSeparateProcess
	 * @preserveGlobalState disabled
	 *
```

Reply:
Oh maybe it's the `preserveGlobalState disabled` that I was missing, b/c when I tried to run this in a separate process I was getting errors about serializing closures.

Reply:
Switched in 783b48aa